### PR TITLE
Added reset-search-input-on attribute with demo.

### PR DESCRIPTION
**Pull request by Sjors**
This allows one to programmatically reset the search field. See #1067.

### CODE REVIEW

File: src/uiSelectController.js [hunk 1]
```diff
@@ -49,8 +49,8 @@ uis.controller('uiSelectCtrl',
   };
 
   // Most of the time the user does not want to empty the search input when in typeahead mode
-  function _resetSearchInput() {
-    if (ctrl.resetSearchInput || (ctrl.resetSearchInput === undefined && uiSelectConfig.resetSearchInput)) {
+  function _resetSearchInput(force) {
+    if (force | ctrl.resetSearchInput || (ctrl.resetSearchInput === undefined && uiSelectConfig.resetSearchInput)) {
```
Comment (wesleycho) [inline src/uiSelectController.js]: Did you mean `||`?

File: src/uiSelectDirective.js [hunk 2]
```diff
@@ -132,6 +132,15 @@ uis.directive('uiSelect',
               });
           });
         }
+        
+        //Reset form based on scope event name (e.g. reset-search-input-on='SomeEventName')
+        if (angular.isDefined(attrs.resetSearchInputOn)){
+          scope.$on(attrs.resetSearchInputOn, function() {
+              $timeout(function(){
```
Comment (wesleycho) [inline src/uiSelectDirective.js]: This seems fishy - why not just do `scope.$apply(function() { $select.clearSearchInput(); });`?

**Comment by wesleycho** [comment]
This needs tests.

**Comment by user378230** [comment]
Closing as don't think this would be more bloat than use. 

Willing to accept a PR to expose a way to reset the search input using $select if necessary. User's can then extend on this with a custom directive ([Accessing $select](https://github.com/angular-ui/ui-select/wiki/Accessing-%24select))

**Pull request closed without merging**